File: uLab/Assets/Scripts/Manager/UIManager.cs


using Locke.ui;


namespace Locke
{

	public class UIManager : Manager
	{
		WindowManager mWindowMgr = null;

		public override void Initialize()
		{
			mWindowMgr = new WindowManager();
			mWindowMgr.Init();
		}

		public override void Destroy()
		{
			this.Cleanup();
		}

		public IWindow GetWindow(string windowName)
		{
			WindowInfo info = this.GetWindowInfo(windowName);
			if (info != null)
				return mWindowMgr.GetWindow(info);
			return null;
		}

		public IWindow OpenWindow(string windowName)
		{
			WindowInfo info = this.GetWindowInfo(windowName);
			if (info != null)
				return this.OpenWindow(info, null);
			else
				return null;
		}

		public IWindow OpenWindow(WindowInfo windowInfo, IContext context = null)
		{
			return mWindowMgr.OpenWindow(windowInfo, context);
		}

		public void CloseWindow(string windowName)
		{
			WindowInfo info = this.GetWindowInfo(windowName);
			if (info != null)
				mWindowMgr.CloseWindow(info);
		}

		public void CloseWindow(IWindow script)
		{
			mWindowMgr.CloseWindow(script);
		}

		public void SetMainWindow(WindowInfo windowInfo)
		{
			mWindowMgr.mainWindowInfo = windowInfo;
		}

		public void BackToMainWindow()
		{
			mWindowMgr.BackToMainWindow();
		}

		public void Cleanup()
		{
			mWindowMgr.Cleanup();
		}

		private WindowInfo GetWindowInfo(string windowName)
		{
			WindowInfo info = GameUI.GetWindowInfo(windowName);
			if (info == null)
			{
				Log.Error(string.Format("Cannot find WindowInfo named : %s", windowName));
			}
			return info;
		}

	}

}

File: uLab/Assets/Lua/UI/UIManager.lua

UIManager = class('UIManager')

function UIManager:ctor()
	self.windowTable = {}
end

function UIManager:OpenWindow(name)
	if not self.windowTable[name] then
		self.windowTable[name] = require('UI/'..name).new()
	end
	Lite.App.uiManager:OpenWindow(name)
end

function UIManager:CloseWindow(name)
	self.windowTable[name] = nil
	Lite.App.uiManager:CloseWindow(name)
end

function UIManager:SetMainWindow(name)
	Lite.App.uiManager:SetMainWindow(name)
end

function UIManager:BackToMainWindow()
	Lite.App.uiManager:BackToMainWindow()
end




